test/suite/test_alter04.py

```diff
@@ -0,0 +1,209 @@
+    ]
+    scenarios = make_scenarios(types, sizes, reopen)
+
+    def verify_metadata(self, metastr):
+        if metastr == '':
+            return
+        cursor = self.session.open_cursor('metadata:', None, None)
+        #
+        # Walk through all the metadata looking for the entries that are
+        # the file URIs for components of the table.
+        #
+        found = False
+        while True:
+            ret = cursor.next()
+            if ret != 0:
+                break
+            key = cursor.get_key()
+            check_meta = ((key.find("lsm:") != -1 or key.find("file:") != -1) \
+                and key.find(self.name) != -1)
+            if check_meta:
+                value = cursor[key]
+                found = True
+                self.assertTrue(value.find(metastr) != -1)
+        cursor.close()
+        self.assertTrue(found == True)
+
+    # Alter: Change the os_cache_max setting after creation
+    def test_alter04_cache(self):
+        uri = self.uri + self.name
+        create_params = 'key_format=i,value_format=i,'
+        complex_params = ''
+        #
+        # If we're not explicitly setting the parameter, then don't
+        # modify create_params to test using the default.
+        #
+        if self.ocreate != '':
+            new_param = 'os_cache_max=%s' % self.ocreate
+            create_params += '%s,' % new_param
+            complex_params += '%s,' % new_param
+        else:
+            # NOTE: This is hard-coding the default value.  If the default
+            # changes then this will fail and need to be fixed.
+            new_param = 'os_cache_max=0'
+
+        cgparam = ''
+        if self.use_cg or self.use_index:
+            cgparam = 'columns=(k,v),'
+        if self.use_cg:
+            cgparam += 'colgroups=(g0),'
+
+        self.session.create(uri, create_params + cgparam)
+        # Add in column group or index settings.
+        if self.use_cg:
+            cgparam = 'columns=(v),'
+            suburi = 'colgroup:' + self.name + ':g0'
+            self.session.create(suburi, complex_params + cgparam)
+        if self.use_index:
+            suburi = 'index:' + self.name + ':i0'
+            self.session.create(suburi, complex_params + cgparam)
+
+        # Put some data in table.
+        c = self.session.open_cursor(uri, None)
+        for k in range(self.entries):
+            c[k+1] = 1
+        c.close()
+
+        # Verify the string in the metadata
+        self.verify_metadata(new_param)
+
+        # Run through all combinations of the alter commands
+        # for all allowed settings.  This tests having only one or
+        # the other set as well as having both set.  It will also
+        # cover trying to change the setting to its current value.
+        for a in self.cache_alter:
+            alter_param = ''
+            if a != '':
+                new_str = 'os_cache_max=%s' % a
+                alter_param += '%s' % new_str
+                if alter_param != '':
```

Comment:
alter_param will never be ' ' here, since it will be at least 'os_cache_max=<something>. Am is missing something ?